cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -49,12 +49,26 @@ public static void abortQuietly(TransactionSystemClient txClient, Transaction tx
 
   public static void invalidateQuietly(TransactionSystemClient txClient, Transaction tx) {
     try {
-      txClient.invalidate(tx.getWritePointer());
+      if (!txClient.invalidate(tx.getWritePointer())) {
+        LOG.error("Failed to invalidate transaction {}", tx);
+      }
     } catch (Throwable t) {
       LOG.error("Exception when invalidating transaction {}", tx, t);
     }
   }
 
+  /**
+   * Wraps the given {@link Throwable} as a {@link TransactionFailureException} if it is not already an instance of
+   * {@link TransactionFailureException}.
+   */
+  public static TransactionFailureException asTransactionFailure(Throwable t) {
```

Comment:
The idea of this method is purely to wrap the given `Throwable` into a `TransactionFailureException` if it is not an instance of `TransactionFailureException` to fit in scenario where only `TransactionFailureException` is supposed to be thrown. The original reason is preserved in the cause of the exception. However, we could have a generic message formed like `"Exception raised inside transaction. Caused: " + t.getMessage()` when wrapping inside a `TransactionFailureException`.